File: CS/GridControlDetailView/CustomGrid/CustomGridControl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Registrator;
using System.Windows.Forms;

namespace CustomGrid
{
    [System.ComponentModel.DesignerCategory("")]
    public class CustomGridControl : GridControl
    {
        protected override void RegisterAvailableViewsCore(InfoCollection collection)
        {
            base.RegisterAvailableViewsCore(collection);
            collection.Add(new Registrator());
        }
    }

    public interface IDetailView
    {
        object DataSource { get; set; }

        Control InternalControl { get; }

        int DetailHeight { get; }
    }
}

File: VB/GridControlDetailView/CustomGrid/CustomGridControl.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Registrator
Imports System.Windows.Forms

Namespace CustomGrid
	<System.ComponentModel.DesignerCategory("")>
	Public Class CustomGridControl
		Inherits GridControl

		Protected Overrides Sub RegisterAvailableViewsCore(ByVal collection As InfoCollection)
			MyBase.RegisterAvailableViewsCore(collection)
			collection.Add(New Registrator())
		End Sub
	End Class

	Public Interface IDetailView
		Property DataSource() As Object

		ReadOnly Property InternalControl() As Control

		ReadOnly Property DetailHeight() As Integer
	End Interface
End Namespace
